### PR TITLE
Add a parameters structure for controlling the embedding options.

### PR DESCRIPTION
**Pull request by greglandrum**

### CODE REVIEW

File: Code/GraphMol/DistGeomHelpers/Embedder.h [hunk 1]
```diff
@@ -221,6 +221,70 @@ INT_VECT EmbedMultipleConfs(
     bool enforceChirality = true, bool useExpTorsionAnglePrefs = false,
     bool useBasicKnowledge = false, bool verbose = false,
     double basinThresh = 5.0);
+
+struct EmbedParameters {
+  unsigned int maxIterations;
```
Comment (bp-kelley) [inline Code/GraphMol/DistGeomHelpers/Embedder.h]: Java and C# will thank you if you just go ahead and add the getters and setters to the struct.  You'll have to write them anyway.

File: Code/GraphMol/DistGeomHelpers/Embedder.h [hunk 2]
```diff
@@ -221,6 +221,70 @@ INT_VECT EmbedMultipleConfs(
...
+        optimizerForceTol(1e-3),
+        ignoreSmoothingFailures(false),
+        enforceChirality(true),
+        useExpTorsionAnglePrefs(false),
+        useBasicKnowledge(false),
+        verbose(false),
+        basinThresh(5.0),
+        pruneRmsThresh(-1.0){};
+};
+inline int EmbedMolecule(ROMol &mol, const EmbedParameters &params) {
```
Comment (bp-kelley) [inline Code/GraphMol/DistGeomHelpers/Embedder.h]: An awesome thing you can do with structs is make common settings for standard ops.

i.e.

```
const EmbedParameters ETDKG = {.....};
```

So people like me with bad memories have a reference.

File: Code/GraphMol/DistGeomHelpers/Embedder.h [hunk 3]
```diff
@@ -221,6 +221,70 @@ INT_VECT EmbedMultipleConfs(
...
+  bool ignoreSmoothingFailures;
+  bool enforceChirality;
+  bool useExpTorsionAnglePrefs;
+  bool useBasicKnowledge;
+  bool verbose;
+  double basinThresh;
+  double pruneRmsThresh;
+  EmbedParameters()
+      : maxIterations(0),
+        numThreads(1),
```
Comment (bp-kelley) [inline Code/GraphMol/DistGeomHelpers/Embedder.h]: Do we have a sentinel (i.e. -1) for use all the threads?  I forget...  Might be useful to have a constant somewhere if so.  i.e. 

```
embedParams.numThreads = ALL_CORES;
```

File: Code/GraphMol/DistGeomHelpers/Embedder.h (hunk 2 above)
Comment (greglandrum) [inline Code/GraphMol/DistGeomHelpers/Embedder.h]: Nice idea, I like this.

File: Code/GraphMol/DistGeomHelpers/Embedder.h (hunk 1 above)
Comment (greglandrum) [inline Code/GraphMol/DistGeomHelpers/Embedder.h]: hmm, looking at the way the AdjustQueryParameters is handled, this doesn't seem to be true. SWIG writes the setters and getters for you. Or did you mean something else?

Comment (bp-kelley) [inline Code/GraphMol/DistGeomHelpers/Embedder.h]: I'll take a look, perhaps newer swig is better in that regards.

Comment (bp-kelley) [inline Code/GraphMol/DistGeomHelpers/Embedder.h]: It looks like there was an issue with vector based setters/getters that don't expose the proxy class that bit me.

http://stackoverflow.com/questions/35203463/swig-getters-setters-for-an-array-of-struct-dont-return-expect-an-array-of-pro

This looks good then.

File: Code/GraphMol/DistGeomHelpers/Embedder.h (hunk 3 above)
Comment (bp-kelley) [inline Code/GraphMol/DistGeomHelpers/Embedder.h]: I'm happy to merge, do you want to add the const definitions now or later?

**Pull request merged**